hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_order_data_manager.py

```diff
@@ -0,0 +1,284 @@
+        self._provider = provider
+        self._owner_address = owner_address
+        self._start_timeout = start_timeout
+
+        for index, trading_pair in enumerate(trading_pairs):
+            trading_pairs[index] = normalize_trading_pair(trading_pair)
+        self._trading_pairs = trading_pairs
+
+        self._order_books: Dict[str, OrderbookInfo] = {}
+        self._markets_to_order_statuses: Dict[str, Dict[int, List[OrderStatusInfo]]] = {}
+
+        self._started = False
+        self._ready = asyncio.Event()
+        self._is_ready = False
+
+        self._orderbook_polling_task: Optional[Task] = None
+        self._order_status_running_tasks: List[Task] = []
+        self._order_status_polling_task: Optional[Task] = None
+
+    async def ready(self):
+        await self._ready.wait()
+
+    @property
+    def is_ready(self):
+        return self._is_ready
+
+    @property
+    def started(self):
+        return self._started
+
+    async def start(self):
+        if not self._started:
+            self._started = True
+            await self._initialize_order_books()
+            self._orderbook_polling_task = asyncio.create_task(self._poll_order_book_updates())
+
+            await self._initialize_order_status_streams()
+
+            await asyncio.sleep(self._start_timeout)
+
+            self._is_ready = True
+            self._ready.set()
+
+    async def stop(self):
+        await self._provider.close()
+        if self._orderbook_polling_task is not None:
+            self._orderbook_polling_task.cancel()
+            self._orderbook_polling_task = None
+        for task in self._order_status_running_tasks:
+            if task is not None:
+                task.cancel()
+        self._order_status_running_tasks.clear()
+
+    async def _initialize_order_books(self):
+        await self._provider.connect()
+        for trading_pair in self._trading_pairs:
+            initialized = False
+            for i in range(5):
+                blxr_orderbook = await self._provider.get_orderbook(market=trading_pair, limit=5, project=OPENBOOK_PROJECT)
+                if blxr_orderbook.market != "":
+                    self._apply_order_book_update(blxr_orderbook)
+                    initialized = True
+
+                    break
+            if not initialized:
+                raise Exception(f"orderbook for {trading_pair} not initialized")
+
+    async def _initialize_order_status_streams(self):
+        for trading_pair in self._trading_pairs:
+            self._markets_to_order_statuses.update({trading_pair: {}})
+
+            initialize_order_stream_task = asyncio.create_task(
+                self._poll_order_status_updates(trading_pair=trading_pair)
+            )
+            self._order_status_running_tasks.append(initialize_order_stream_task)
+
+    async def _poll_order_book_updates(self):
+        await self._provider.connect()
```

Comment:
avoid these repeated calls to `connect`. Connecting should only be done in a single spot. If you're wait for the ready status, then `await self._ready.wait()`